### ISSUE TITLE
[BUG] SessionManager: saveLatestOn 'trigger' with no snapshotTrigger silently never saves

### ISSUE THREAD

**Issue by pgrayy** [strands-agents/sdk-typescript#725]
### Checks

- [x] I have updated to the latest minor and patch version of Strands
- [x] I have checked the documentation and this is not expected behavior
- [x] I have searched ./issues and there are no duplicates of my issue

### Strands Version

0.7.0

### Node.js Version

22.x

### Operating System

macOS

### Installation Method

npm

### Steps to Reproduce

```typescript
import { SessionManager, FileStorage } from "@strands-agents/sdk"

const session = new SessionManager({
  sessionId: "my-session",
  storage: { snapshot: new FileStorage() },
  saveLatestOn: "trigger",
  // no snapshotTrigger provided
})

const agent = new Agent({ sessionManager: session })
await agent.invoke("Hello")
// snapshot_latest is never saved, conversation state is silently lost
```

### Expected Behavior

The `SessionManager` constructor should throw a validation error when `saveLatestOn` is set to `"trigger"` but no `snapshotTrigger` callback is provided. This is a misconfiguration that leads to silent data loss.

### Actual Behavior

No error is thrown. The session manager initializes successfully but never saves `snapshot_latest` because the trigger check in `_onAfterAgentInvocation` short-circuits:

```typescript
if (this._snapshotTrigger?.({ agentData: agent })) {
  await this._saveImmutableAndLatest(agent)
}
```

With no trigger defined, this is always falsy, so no snapshots are ever persisted.

### Additional Context

Found during bug bash by @maczas.

### Possible Solution

Add validation in the constructor:

```typescript
if (this._saveLatestOn === "trigger" && !this._snapshotTrigger) {
  throw new Error("snapshotTrigger is required when saveLatestOn is \"trigger\"")
}
```

### Related Issues

N/A
